### PR TITLE
feat: compressed token transfers

### PR DESCRIPTION
**Pull request by ananas-block**

### CODE REVIEW

File: zk.js/src/idls/psp_compressed_pda.ts [hunk 1]
```diff
@@ -1121,6 +1180,41 @@ export const IDL: PspCompressedPda = {
...
+            "type": {
+              "array": [
+                "u8",
+                32
+              ]
+            }
+          }
+        ]
+      }
+    },
```
Comment (SwenSchaeferjohann) [inline zk.js/src/idls/psp_compressed_pda.ts:1217]: seems there's still `ProofCompressed` as well, which afaik has .a .b .c. 

can we remove one of them?

**Review by SwenSchaeferjohann — COMMENTED** [review]
minor comments, 
there's a "proofcompressed" struct in the verifier SDK; if not remove it already (preferred) i would recommend adding a todo: remove comment to the struct.

i think in general for anything that we think we'd want to review again before prod / consider changing,  we should add inline comments with todos; for example names of structs or methods where its unclear that its a decent shape yet.

File: zk.js/src/idls/psp_compressed_pda.ts (hunk 1 above)
Comment (ananas-block) [reply zk.js/src/idls/psp_compressed_pda.ts:1217]: We are moving away from verifier sdk. Changed the CompressedProof struct in compressed-pda utils.rs

File: programs/compressed-token/src/process_transfer.rs [hunk 2]
```diff
@@ -1,183 +1,264 @@
...
+        let tlv = Tlv {
+            tlv_elements: vec![tlv_data],
+        };
+        in_utxo.in_utxo.data = Some(tlv);
+    }
+    Ok(())
+}
+
+#[inline(never)]
+pub fn cpi_create_execute_compressed_instruction_transfer<'info>(
```
[the remark this answers is not in the document]
Comment (ananas-block) [reply programs/compressed-token/src/process_transfer.rs:96]: It's being called once and is very descriptive, if you don't have a suggestion I am inclined to leave it as is.

[the remark this answers is not in the document]
Comment (SwenSchaeferjohann) [reply programs/compressed-token/src/process_transfer.rs:96]: how about
`cpi_create_and_execute_transfer_instruction`
or
`cpi_create_and_execute_compressed_transfer_instruction`
?

[the remark this answers is not in the document]
Comment (ananas-block) [reply programs/compressed-token/src/process_transfer.rs:96]: It doesn't execute anything it just creates the instruction.

[the remark this answers is not in the document]
Comment (ananas-block) [reply programs/compressed-token/src/process_transfer.rs:96]: cpi_execute_compressed_transaction_transfer

it cpis the compressed_pda program's execute_compressed_transaction instruction for the compressed token program's transfer instruction.

[the remark this answers is not in the document]
Comment (ananas-block) [reply programs/compressed-token/src/process_transfer.rs:96]: renamed the corresponding function for mint_to

File: programs/compressed-token/src/process_mint.rs
Comment (SwenSchaeferjohann) [inline programs/compressed-token/src/process_mint.rs:1]: nit: `create_initiatialize_mint_instruction` typo -> `create_initialize_mint_instruction`

File: js/stateless.js/src/idls/psp_compressed_token.ts
Comment (SwenSchaeferjohann) [inline js/stateless.js/src/idls/psp_compressed_token.ts:1]: cool lgtm, just need to add an interim `CompressedProof` and `InUtxoTuple` to make ts program() compile in client

**Review by SwenSchaeferjohann — COMMENTED** [review]
>

**Review by SwenSchaeferjohann — APPROVED** [review]

**Pull request merged**